arm9/src/unk_0204AEA8.c

```diff
@@ -0,0 +1,45 @@
+#include "global.h"
+#include "heap.h"
+#include "unk_0204639C.h"
+#include "unk_0204AEA8.h"
+
+#pragma thumb on
+
+extern void MOD05_021E3444(u32, struct UnkStruct_0204639C *, u32);
+extern void FUN_0200433C(u32, u32, u32);
+
+BOOL FUN_0204AEA8(struct UnkStruct_0204639C *a0)
+{
+    struct UnkStruct_0204639C *v0 = FUN_02046528(a0);
+    u32 *v1 = FUN_0204652C(a0);
+
+    switch (v1[0])
+    {
+        case 0:
+            MOD05_021E3444(v1[2], v0, (u32)v1 + 4);
+            FUN_0200433C(5, ((v1[3] << 16) >> 16), 1);
```

Comment:
a bitshift left follow by a bitshift right by 16 is a cast to u16, please replace